akka-docs-dev/rst/java/http/routing-dsl/testkit.rst

```diff
@@ -0,0 +1,80 @@
+
+
+Writing Asserting against the HttpResponse
+------------------------------------------
+
+The testkit supports a fluent DSL to write compact assertions on the response by chaining assertions
+using "dot-syntax". To simplify working with streamed responses the entity of the response is first "strictified", i.e.
+entity data is collected into a single ``ByteString`` and provided the entity is supplied as an ``HttpEntityStrict``. This
+allows to write several assertions against the same entity data which wouldn't (necessarily) be possible for the
+streamed version.
+
+All of the defined assertions provide HTTP specific error messages aiding in diagnosing problems.
+
+Currently, these methods are defined on ``TestResponse`` to assert on the response:
+
+=================================================================== =======================================================================
+Assertion                                        Description
+=================================================================== =======================================================================
+``assertStatusCode(int expectedCode)``                              Asserts that the numeric response status code equals the expected one
+``assertStatusCode(StatusCode expectedCode)``                       Asserts that the response ``StatusCode`` equals the expected one
+``assertMediaType(String expectedType)``                            Asserts that the media type part of the response's content type matches
+                                                                    the given String
+``assertMediaType(MediaType expectedType)``                         Asserts that the media type part of the response's content type matches
+                                                                    the given ``MediaType``
+``assertEntity(String expectedStringContent)``                      Asserts that the entity data interpreted as UTF8 equals the expected
+                                                                    String
+``assertEntityBytes(ByteString expectedBytes)``                     Asserts that the entity data bytes equal the expected ones
+``assertEntityAs(Unmarshaller<T> unmarshaller, expectedValue: T)``  Asserts that the entity data if unmarshalled with the given marshaller
+                                                                    equals the given value
+``assertHeaderExists(HttpHeader expectedHeader)``                   Asserts that the response contains an HttpHeader instance equal to the
+                                                                    expected one
+``assertHeaderKindExists(String expectedHeaderName)``               Asserts that the response contains a header with the expected name
+``assertHeader(String name, String expectedValue)``                 Asserts that the response contains a header with the given name and
+                                                                    value.
```

Comment:
on one hand I like this table, on the other one seems to overlap with plain scaladoc a lot. Let's keep for now it's good for people learning it for the first time